docs/data/material/migration/upgrade-to-v7/upgrade-to-v7.md

```diff
@@ -160,6 +160,68 @@ npx @mui/codemod@next v7.0.0/input-label-size-normal-medium <path/to/folder>
 
 The default `data-testid` prop has been removed from the icons in `@mui/icons-material` in production bundles. This change ensures that the `data-testid` prop is only defined where needed, reducing the potential for naming clashes and removing unnecessary properties in production.
 
+### Theme behavior changes
+
+When CSS theme variables is enabled with built-in light and dark color schemes:
+
+```js
+const theme = createTheme({
+  cssVariables: {
+    colorSchemeSelector: 'class',
+  },
+  colorSchemes: {
+    light: true,
+    dark: true,
+  },
+});
+console.log(theme.palette.mode); // 'light'
+```
+
+The theme no longer changes between modes:
+
+```js
+const { setMode } = useColorScheme();
+setMode('dark');
+console.log(theme.palette.mode); // 'light'
```

Comment:
How would someone implement a color mode toggler if they can't use `useColorScheme` to get the current mode? 🤔 It's extremely confusing doing `setMode('dark')` and not getting the updated value.

---

Off-topic: I just realized the hook is named `useColorScheme` but it returns `mode` and `setMode`, why is that?

Reply:
Those are separate thing, the `mode` you get from `useColorScheme` is a React state that ThemeProvider created internally.

When the `mode` state changes, ThemeProvider recalculate the theme as pass to Emotion theme context, causing all components to rerender (this is the previous behavior).

> It's extremely confusing doing setMode('dark') and not getting the updated value.
This is not true, because when you call `setMode('dark')` the `mode` state always update. It's just that the theme now does not change.

May be a blog post or a doc about it might help. I think moving from JS theme to CSS vars theme is quite a shift, I will try to draft the doc about the transition.

Reply:
If I want to build a light/dark toggle using `useColorScheme` or if I want to have conditional rendering based on the current color scheme (mode), how can I do it if the `mode` returned from `useColorScheme` doesn't return the updated value?

Reply:
The `mode` returned from `useColorScheme` always updated (it's a React state). It's independent from the theme.

Reply:
I think there's a typo then? I was confused by the code snippet.

```suggestion
const { setMode } = useColorScheme();
setMode('dark');
console.log(theme.palette.mode); // 'dark'
```

Reply:
I mean, the snippet is correct (the logged value will be `'light'` until the component re-renders) but it's confusing, especially in conjunction with the phrase preceding the code snippet ("The theme no longer changes between modes"). Can we change the snippet and the preceding phrase?

Reply:
I see, it's confusing indeed. I updated the snippet and add more details to it. The key here is to show that `mode` from `useColorScheme` is always updated with the setter but the theme object does not change between modes.